### DISCUSSION TITLE
pre-handler hook

### DISCUSSION THREAD

**Discussion by vjau** [fastify/fastify#3219] [Q&A]
What does the "pre-handler" hook relates to ? The documentation is mute about this. Thank you.

**Comment by RafaelGSS** [comment]
Basically, it's a handler that is executed before the main handler itself. You can see the lifecycle here: https://www.fastify.io/docs/latest/Lifecycle/

**Reply by vjau** [reply]
Thank you, what does the handler do ? Handle routes ?

**Reply by RafaelGSS** [reply]
Handle incoming requests. It's up to you to define what this handle will do, usually, it's used as common `middleware`.
